### PR TITLE
[ui] Use single hidden asset job for all single-run asset executions

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

Related: https://linear.app/dagster-labs/issue/FE-509/update-the-materialize-button-to-pull-jobpartitiondefinition-using-the

This PR stacks on top of #23494:

- On the backend, there is now just a single hidden asset job which contains all assets.

- This asset job has no partition_set, but given a set of asset keys, you can still retrieve the list of available partitions and config + tags for a specific partition using new resolvers on the job. (Added in @sryza's PR)

This requires some changes to expectations made on the front-end:

- We no longer check both the partition definitions AND the presence of a partition set on the job the assets have in common. If you select partitioned assets you will get the partitions dialog.

- When we launch a job targeting a single partition, OR open the launchpad and allow you to choose a partition, we now use a new resolver that is asset-specific to retrieve the config YAML + tags. (The "partition => config YAML" resolver previously required the partition set definition)

The last point means that there is a different config-loader query for the hidden asset job and standard user-defined asset/  op jobs (which still use partition set definitions).  I co-located both approaches in the same file to try to make things a bit easier to follow, but there's a bit of code duplication.

Sidenote:

This code is super old and still supports pipelines with multiple modes.  If modes are fully gone I would love to rip that stuff out...

## How I Tested These Changes

I updated the tests and also ran through a bunch of manual tests listed that I listed out in https://www.notion.so/dagster/Hidden-asset-job-change-test-cases-7a96046234da46f3a0e6b2a44e349474?pvs=4.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/app/ExecutionSessionStorage.tsx [hunk 1]
```diff
@@ -31,8 +31,20 @@ export interface PipelineRunTag {
...
+  | {
+      isAssetJob: true; // partition set name not required
+      partitionName: string | null;
+      tags: PipelineRunTag[] | null;
+    }
+  | {
+      partitionsSetName: string; // required for op jobs
+      partitionName: string | null;
+      tags: PipelineRunTag[] | null;
+    };
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/app/ExecutionSessionStorage.tsx:47]: Might be nice to give these types individual names just to clarify their purposes.

Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/app/ExecutionSessionStorage.tsx:47]: Ahh yeah I like that, will add that instead of the isAssetJob key

**Comment by sryza** [comment]
@bengotow @salazarm do you think we can get this into this week's release?

File: js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigFetch.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  client: ApolloClient<any>,
+  variables: ConfigPartitionSelectionQueryVariables,
+) {
+  const {data} = await client.query<
+    ConfigPartitionSelectionQuery,
+    ConfigPartitionSelectionQueryVariables
+  >({
+    query: CONFIG_PARTITION_SELECTION_QUERY,
+    variables,
+  });
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigFetch.tsx:23]: hmm, this should really be considered as part of the page load imo if it's part of the launchpad page. Not a blocker for this PR but I'm thinking maybe we just have some kind of wrapper around async functions

**Review by salazarm — APPROVED** [review]
makes sense to me

File: js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigFetch.tsx (hunk 2 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigFetch.tsx:23]: Ahh this one might be tricky... I think for op jobs the launchpad loads and then you have to choose a partition before this config is fetched -- in some cases I think it does autopopulate (for presets maybe?) but not all the time.

**Pull request merged**